### DISCUSSION TITLE
Question: What is the difference between ngx-editor and ProseMirror?

### DISCUSSION THREAD

**Discussion by vella-nicholas** [sibiraj-s/ngx-editor#339] [Q&A] [answered]
In the documentation, I noticed a lot of examples leading to the ProseMiror project. What is the relation between these two projects?

**Comment by sibiraj-s** [comment, marked as answer]
ProseMirror is a toolkit providing the raw APIs for building a rich content editor. This editor is built on top of it. You can also think of `ngx-editor` as angular bindings for ProseMirror.

The editor provides some useful defaults, commands, and few UI components. like menu, floating menu. 

The editor is simple by default. But there are situations one might need some advanced functionalities, which can be added by providing Prosemirror plugins found in the NPM registry or a custom one.

**Reply by vella-nicholas** [reply]
Can all the functionality featured in ProseMirror be done in ngx-editor? I read the documentation but I like to be sure.
Should I need to use ProseMirror in angular, this should be my go-to project right? :)

**Reply by sibiraj-s** [reply]
> Can all the functionality featured in ProseMirror be done in ngx-editor?

Yes. You can.

> Should I need to use ProseMirror in angular, this should be my go-to project right?

The answer is it depends. It can be a goto project. since it is easy to start and simple to setup. Everything required for a simple/medium complex editor is there I think. May be for complex uses one can prefer tiptap v2 (published recently, haven't checked how that works, you may need to still write the angular bindings)

This library provides some useful defaults like.
- form/ng-model bindings
- reactive forms support.
- form validations, custom validations can be added
- basic commands to programatically work with editor
- a menu component.
- a simple floating menu
- additionally examples for few use cases, and more added based on user requests.

One can also build Prosemirror directly to your angular projects If they know Prosemirror well.

**Reply by vella-nicholas** [reply]
Thank you so much for your assistance on the matter!

[a comment by an automated account is not shown]
